client/src/api/ApiClient.ts

```diff
@@ -0,0 +1,28 @@
+export interface ApiClient {
+  authenticate(username: string, password: string): Promise<boolean>
+}
+
+// TODO: Add CSRF protection
+export class ApiClientImpl {
+  private baseRoute = '/api';
+
+  authenticate(username: string, password: string): Promise<boolean> {
+    return fetch(`${this.baseRoute}/authenticate`, {
+      method: 'POST',
+      cache: 'no-cache',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        username: encodeURIComponent(username),
```

Comment:
because I was trying to show I understand security against injecting json escape sequences into the input fields, but then I forgot that JSON.stringify prevents that anyway.